tests/unit/amo/pages/TestSearchPage.js

```diff
@@ -153,6 +157,28 @@ describe(__filename, () => {
     sinon.assert.notCalled(fakeDispatch);
   });
 
+  it('dispatches a server redirect when `category` and `addonType` is set', () => {
+    const category = 'some-category';
+    const page = '123';
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    render({
+      location: createFakeLocation({
+        query: { category, page, type: ADDON_TYPE_EXTENSION },
+      }),
+      store,
+    });
+
+    sinon.assert.callCount(fakeDispatch, 1);
+    sinon.assert.calledWith(
+      fakeDispatch,
+      sendServerRedirect({
+        status: 301,
+        url: `/en-US/android/extensions/category/${category}/?page=${page}`,
```

Comment:
Maybe explicitly pass the `lang` or `clientApp` in this test case to make sure the values are updated correctly.